src/devices/Ft4222/FtCommon.cs

```diff
@@ -24,23 +25,33 @@ public static List<DeviceInformation> GetDevices()
             FtStatus ftStatus = 0;
 
             // Check device
-            uint numOfDevices = 0;
-            ftStatus = FtFunction.FT_CreateDeviceInfoList(ref numOfDevices);
+            uint numOfDevices;
+            ftStatus = FtFunction.FT_CreateDeviceInfoList(out numOfDevices);
 
             Debug.WriteLine($"Number of devices: {numOfDevices}");
             if (numOfDevices == 0)
                 throw new IOException($"No device found");
 
-            byte[] sernum = new byte[16];
-            byte[] desc = new byte[64];
-            uint flags = 0;
-            for (int i = 0; i < numOfDevices; i++)
+            for (uint i = 0; i < numOfDevices; i++)
             {
+                Span<byte> sernum = stackalloc byte[16];
+                Span<byte> desc = stackalloc byte[64];
+                uint flags = 0;
                 var devInfo = new DeviceInformation();
-                ftStatus = FtFunction.FT_GetDeviceInfoDetail(0, ref flags, ref devInfo.Type, ref devInfo.Id, ref devInfo.LocId, sernum, desc, ref devInfo.FtHandle);
+                FtDevice ftDevice;
+                uint id;
+                uint locId;
+                IntPtr handle;
+                ftStatus = FtFunction.FT_GetDeviceInfoDetail(i, out flags, out ftDevice, out id, out locId , out MemoryMarshal.GetReference(sernum), out MemoryMarshal.GetReference(desc), out handle);
+                if (ftStatus != FtStatus.Ok)
+                    throw new IOException($"Can't read device information on device index {i}, error {ftStatus}");
+                devInfo.Type = ftDevice;
+                devInfo.Id = id;
+                devInfo.LocId = locId;
+                devInfo.FtHandle = handle;
                 devInfo.Flags = (FtFlag)flags;
-                devInfo.SerialNumber = Encoding.ASCII.GetString(sernum, 0, 16);
-                devInfo.Description = Encoding.ASCII.GetString(desc, 0, 64);
+                devInfo.SerialNumber = Encoding.ASCII.GetString(sernum.ToArray(), 0, 16);
```

Comment:
@GrabYourPitchforks do you perhaps if there is any way to do that without going through array?

Also since this is returning string I'm unsure if this will take care of the null terminator correctly